### ISSUE TITLE
V1.7.1 'Publish Local Workspace' now not working. V1.7.0 still works

### ISSUE THREAD

**Issue by ianoel230982** [chrishasz/spgo#146]
**Note: for support questions, please use stackoverflow and tag the post with #spgo**.

* **I'm submitting a ...**

  * [x ] bug report

* **What is the current behavior?**
* VS Code has auto updated my extension to the new version 1.7.1
* Since this version i can no longer 'Publish Local Workspace' without receiving an error in the VSCode output
* Error Message in the terminal output is below
[13/05/2021, 16:51:40] ================================     ERROR     ================================
[13/05/2021, 16:51:40] EPERM: operation not permitted, scandir 'C:/$Recycle.Bin/S-1-5-18'
[13/05/2021, 16:51:40] Error Detail:
[13/05/2021, 16:51:40] ----------------------
[13/05/2021, 16:51:40] Error: EPERM: operation not permitted, scandir 'C:/$Recycle.Bin/S-1-5-18'
[13/05/2021, 16:51:40] ===============================================================================

* **If the current behavior is a bug, please**
  
  * Your `spgo.json` file (You may replace corporate PII with "xxx")
  *
```
 {
	"sourceDirectory": "src",
	"sharePointSiteUrl": "http://siteURL/sites/SiteName",
	"publishingScope": "SaveOnly",
	"authenticationType": "NTLM",
	"publishWorkspaceOptions": {
		"destinationFolder": "Web/",
		"localRoot": "/src/Development/Web",
		"globPattern": "/**/*.*"
	},
	"workspaceRoot": "c:\\SPGo\\RootFolderName",
	"checkInMessage": "Updated by Me",
	"remoteFolders": ["/Web/**/"],
	"subSites": [
		{
			"sharePointSiteUrl": "http://siteURL/sites/SiteName/Development",
			"remoteFolders": ["/Web/**/"]
		}
	]
}
```
  * Steps to reproduce
  * Press F1 in Vs Code, choose 'Publish Local Workspace', enter the check-in message and press enter. Error message appears in terminal output

* **What is the expected behavior?**
* Publish all files in the local root to the destination folder
* *_I have regressed to previous version 1.7.0 and the correct behaviour expected occurs._*

* **Please tell us about your environment:**
  
  * OS and Version - Windows 8.1
  * SharePoint Location (2013 On Premise)
  * Authentication Type - NTFS

**Comment by chrishasz** [comment]
Thank you, @ianoel230982. I think I know the root cause of this error.

As of 1.7.1, I made the parser much more explicit regarding absolute paths. Since you are using `/src/development/web`, and `/**/*.*` in the `publishWorkspaceOptions` node, SPGo is interpreting this to mean "start at the root of the file system and look for the path `/src/development/web`.

Can you remove the leading '/' character from the `localRoot` and `globPattern` properties?

That said, since the config file you use has web-style slashes, I did find a bug translating those to windows slashes. I'll have a fix up on the marketplace today.

**Comment by chrishasz** [comment]
I've published SPGo 1.7.2 to the VSCode Marketplace. Once you make the changes I listed in the message above to your spgo.json file, workspace publish should work again for you.

**Comment by ianoel230982** [comment]
Hi,

Thanks for your quick turn around on this. I now face another problem, which i'll try and explain below.

I've tried installing the new v1.7.2 extension via the market place but it was not compatible with the version of vs code i was running 1.54.3

I have updated VSCode to the latest version 1.56.2 and installed the latest version of the extension you published last night but now i have another error that is stopping me populating the workspace or carrying out any SPGo actions.

The error i now received in the terminal output is
[14/05/2021, 09:36:43] ================================     ERROR     ================================
[14/05/2021, 09:36:43] 401 - undefined
[14/05/2021, 09:36:43] Error Detail:
[14/05/2021, 09:36:43] ----------------------
[14/05/2021, 09:36:43] StatusCodeError: 401 - undefined
[14/05/2021, 09:36:43] ===============================================================================

A colleague of mine is also receiving the same error since updating VScode to the latest version.

Is there any possible advice you could provide as to this?

I'll regress VSCode and the SPGo extension to what is currently working for me, but will certainly be happy to try any suggestions you have?

**Comment by chrishasz** [comment]
Do you see this with every command, or only publish workspace?

Can you post the SPGo.json file you were using?

**Comment by ianoel230982** [comment]
Hi,

Its the same SPGo.json file as above. Do i need to try it by removing the '/' as suggested to fix the other issue?

I tried the Populate Workspace command which asked for my credentials, I entered my credentials in format of 'domain\username' and then my password, on hitting 'enter' the error message appears in the output.
I can't try any other commands as it just asks for my credentials each time, which keeps failing with the above error.

**Comment by sg-chrishasz** [comment]
Yes, you will need to remove the leading slash, as per the comment above.

Having a leading slash tells SPGo “start at the root” just like having “c:/“ would.

**Comment by ianoel230982** [comment]
Thanks, i'll give that a go and report back in a little while.

**Comment by ianoel230982** [comment]
OK, so i am now at the following

Version: 1.56.2 (system setup)
Commit: 054a9295330880ed74ceaedda236253b4f39a335
Date: 2021-05-12T17:13:13.157Z
Electron: 12.0.4
Chrome: 89.0.4389.114
Node.js: 14.16.0
V8: 8.9.255.24-electron.0
OS: Windows_NT x64 6.3.9600

SPGo V1.7.2

My SPGo.json file is as below

```
 {
	"sourceDirectory": "src",
	"sharePointSiteUrl": "http://siteURL/sites/SiteName",
	"publishingScope": "SaveOnly",
	"authenticationType": "NTLM",
	"publishWorkspaceOptions": {
		"destinationFolder": "Web/",
		"localRoot": "src/Development/Web",
		"globPattern": "**/*.*"
	},
	"workspaceRoot": "c:\\SPGo\\RootFolderName",
	"checkInMessage": "Updated by Me",
	"remoteFolders": ["/Web/**/"],
	"subSites": [
		{
			"sharePointSiteUrl": "http://siteURL/sites/SiteName/Development",
			"remoteFolders": ["/Web/**/"]
		}
	]
}
```

As soon as i try and do _any_ SpGo command i am prompted with the credentials, to which i use my domain\username and password. I still receive the same 401 error in the output.

Any thoughts?

**Comment by ianoel230982** [comment]
Hi, further update for this. I came across this issue #73 and noticed it was similar behaviour to what i am now seeing. I have carried out the advice for disabling the proxy support for extensions and i am back up and running. I can even publish the workspace with your update instructions for removing the slash.

With that in mind i will close this bug now, as you have fixed the original issue. I guess the other issue is with VSCode.

Thanks again for your continued support.
